app/models/review.rb

```diff
@@ -0,0 +1,17 @@
+class Review < ActiveRecord::Base
+  after_save :update_video_rating!
+
+  belongs_to :creator, class_name: 'User', foreign_key: 'user_id'
+  belongs_to :video
+
+  validates_presence_of :user_id
+  validates_presence_of :video_id
+  validates_inclusion_of :rating, in: 1..5
+  validates :body, length: { minimum: 10 }
+
+  private
+
+  def update_video_rating!
+    video.save if video
```

Comment:
Here's another option. What if you don't store a rating on the video at all, and you have a method on the `Video` class which calculates it on the fly?